crates/holochain_integrity_types/src/action/builder.rs

```diff
@@ -49,7 +49,7 @@ impl ActionBuilderCommon {
 ///
 /// This builder does not build pre-genesis Actions, because prior to genesis
 /// there is no Agent associated with the source chain, and also the fact that
-/// the Dna action has no prev_entry causes a special case that need not be
+/// the Dna action has no prev_action causes a special case that needs not be
```

Comment:
```suggestion
/// the Dna action has no prev_action causes a special case that need not be
```

Need was actually correct in this sentence

Reply:
Is this clearly a subjunctive? =) May well be.

Reply:
Eww, grammar :) If it was the positive case then needs would be correct "needs to be", but for the negative it should be "need not be". I'm afraid I figure this stuff out by whether it looks right when I read it, my knowledge of formal grammar is shamefully poor :D

Reply:
I'd be in favour of the sentence being re-written to be simpler, then we don't have to work out which it should be at all :D

Reply:
Yeah, that is a gnarly sentence!